test/IISExpress.FunctionalTests/HttpsTests.cs

```diff
@@ -0,0 +1,198 @@
+        }
+
+        [ConditionalTheory]
+        [RequiresIIS(IISCapability.Admin)]
+        [MemberData(nameof(TestVariants))]
+        public Task HttpsHelloWorld_ClientCert(TestVariant variant)
+        {
+            return HttpsHelloWorldCerts(variant, sendClientCert: true);
+        }
+
+        private async Task HttpsHelloWorldCerts(TestVariant variant, bool sendClientCert)
+        {
+            var port = TestPortHelper.GetNextSSLPort();
+            var deploymentParameters = _fixture.GetBaseDeploymentParameters(variant);
+            deploymentParameters.ApplicationBaseUriHint = $"https://localhost:{port}/";
+            deploymentParameters.AddHttpsToServerConfig();
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            var handler = new HttpClientHandler
+            {
+                ServerCertificateCustomValidationCallback = (a, b, c, d) => true,
+                ClientCertificateOptions = ClientCertificateOption.Manual,
+            };
+
+            if (sendClientCert)
+            {
+                var clientCert = FindClientCert();
+
+                if (clientCert == null)
+                {
+                    clientCert = CreateRootCert();
+                }
+                Assert.NotNull(clientCert);
+                handler.ClientCertificates.Add(clientCert);
+            }
+
+            var client = deploymentResult.CreateClient(handler);
+            var response = await client.GetAsync("GetClientCert");
+
+            var responseText = await response.Content.ReadAsStringAsync();
+
+            if (sendClientCert)
+            {
+                Assert.Equal("Enabled", responseText);
+            }
+            else
+            {
+                Assert.Equal("Disabled", responseText);
+            }
+        }
+
+        private X509Certificate2 CreateRootCert()
+        {
+            var parentKey = CreateKeyMaterial(2048);
+
+            // On first run of the test, creates the certificate in the trusted root certificate authorties.
+            var parentRequest = new CertificateRequest("CN=IISIntegrationTest_Root", parentKey, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+
+            parentRequest.CertificateExtensions.Add(
+                new X509BasicConstraintsExtension(
+                    certificateAuthority: true,
+                    hasPathLengthConstraint: false,
+                    pathLengthConstraint: 0,
+                    critical: true));
+
+            parentRequest.CertificateExtensions.Add(
+                new X509KeyUsageExtension(X509KeyUsageFlags.DigitalSignature | X509KeyUsageFlags.NonRepudiation, critical: true));
+
+            parentRequest.CertificateExtensions.Add(
+                new X509SubjectKeyIdentifierExtension(parentRequest.PublicKey, false));
+
+            var notBefore = DateTimeOffset.Now.AddDays(-1);
+            var notAfter = DateTimeOffset.Now.AddYears(5);
```

Comment:
`AddDays(1)` ?

Reply:
Why?

Reply:
Because you expect the cert to be cleaned up at the end of the test.